src/Container.tsx

```diff
@@ -131,22 +131,16 @@ export const Container = React.memo(
         () => tabNamesArray,
         [tabNamesArray]
       )
-      const index: ContextType['index'] = useSharedValue(
-        initialTabName
-          ? tabNames.value.findIndex((n) => n === initialTabName)
-          : 0
-      )
+      const index: ContextType['index'] = useSharedValue(initialIndex)
 
       const focusedTab: ContextType['focusedTab'] =
         useDerivedValue<TabName>(() => {
           return tabNames.value[index.value]
         }, [tabNames])
-      const calculateNextOffset = useSharedValue(index.value)
+      const calculateNextOffset = useSharedValue(initialIndex)
```

Comment:
This change is safe to do because the initial value passed to useSharedValue will only be captured once on first pass, this is not reactive.